mhlib/app.go

```diff
@@ -269,3 +293,81 @@ func (a *App) render(configFile string) (*string, *string, *[]byte, error) {
 
 	return &chart, &chartVersion, &overrides, nil
 }
+
+
+func selfRender(templateValuesStr string, enableGomplate bool) (string, error) {
+	/*
+	This function will accept an input string and run it through the
+	  templating engine as both the values dictionary as well as the
+	  template string.  It will repeat the process until the templating
+	  result stops changing.
+	*/
+
+	type Values map[string]interface{}
+
+	lastRender := templateValuesStr
+	for i := 0; i < 10; i++ {
+
+		// Unmarshal the file as a values dict
+		vals := Values{}
+		err := yaml.Unmarshal([]byte(templateValuesStr), &vals)
+		if err != nil { return "", err }
+
+		tmpl := template.New("SelfTemplate")
+		tmpl.Delims("[[", "]]")
+
+		if enableGomplate {
+			// Read the defined datasources and datasourcehaders
+			//   from the values dict
+			dataSources := []string{}
+			dataSourceHeaders := []string{}
+			if _, ok := vals["gomplate"]; ok {
+				tmp := vals["gomplate"].(map[string]interface{})
+				if _, ok := tmp["datasources"]; ok {
```

Comment:
I think this if block can be
```
if value, ok := vals["gomplate"]; ok {
	if gomp, ok := value.(map[string]interface{}); ok {
		dataSources, _ = gomp["datasources"].([]string)
		dataSourceHeaders, _ = gomp["datasourceheaders"].([]string)
	}
}
```

See the structure here:
https://github.com/hairyhenderson/gomplate/blob/master/gomplate.go

Reply:
Passes:
```
func TestSelfRender(t *testing.T) {
	templateString := `gomplate:
  datasources:
  - "http_obj=https://httpbin.org/get"
  datasourceheaders: []
values:
  a: foo
  b: '[[ .values.a ]]bar' # foobar
  c: '[[ .values.b ]]baz' # foobarbaz
  f: 'Func Test: [[ net.LookupIP "example.com" ]]'
`
	expected := `gomplate:
  datasources:
  - "http_obj=https://httpbin.org/get"
  datasourceheaders: []
values:
  a: foo
  b: 'foobar' # foobar
  c: 'foobarbaz' # foobarbaz
  f: 'Func Test: 10.114.236.103'
`
	out, err := selfRender(templateString, true)
	if err != nil {
		t.Log(err)
		t.Fail()
	}
	if out != expected {
		t.Logf("Actual: %s\nExpected: %s\n", out, expected)
		t.Fail()
	}
}
```

Reply:
I've made the a slight variation of your suggestion, without the extra marshal step.

Also, added your unit test snippet in next commit.